docs: Add documentation for disabling ReclaimSpace (#701)
csi-addons/kubernetes-csi-addons: docs/reclaimspace.md

CSI Addons will not perform any further modifications on the `ReclaimSpaceCronJob` with the `unmanaged` state.

To have a custom schedule the user can then modify the `schedule` field of the `ReclaimSpaceCronJob` spec.

## Disabling Reclaim Space

### Disabling Reclaim Space for a Specific PersistentVolumeClaim

> To disable reclaim space for a specific PersistentVolumeClaim (PVC), edit the `ReclaimSpaceCronJob` custom resource (CR) associated with that PVC. Follow these steps:

Reviewer: ```suggestion
To disable reclaim space for a specific PersistentVolumeClaim (PVC), follow these steps to modify the associated `ReclaimSpaceCronJob` CR:
```

1. **Identify the `ReclaimSpaceCronJob` CR**: Run the following command to retrieve the name of the `ReclaimSpaceCronJob` CR associated with the PVC:

   ```bash
   kubectl get reclaimspacecronjobs -o jsonpath='{range .items[?(@.spec.jobTemplate.spec.target.persistentVolumeClaim=="<PVC_NAME>")]}{.metadata.name}{"\n"}{end}'
   ```

   Replace `<PVC_NAME>` with the name of your PVC.

> 2. **Edit the `ReclaimSpaceCronJob` CR**: Use the following settings in the CR to disable reclaim space for this specific PVC:

Reviewer: ```suggestion
2. **Edit the `ReclaimSpaceCronJob` CR**: apply the following to disable the reclaim space:
```

   - Update the `csiaddons.openshift.io/state` annotation from `"managed"` to `"unmanaged"`.

>    - Add `suspend: true` under the `spec` field.

Author: Done!

>    - Update the `csiaddons.openshift.io/state` annotation from `"managed"` to `"unmanaged"`.

Reviewer: will be good if we add example for kubectl annotate and kubectl patch command

Author: The reason I did not go with that is because

- The annotation would be present on the CR already.
- User needs to just to `kubectl edit` and can do both the required changes simultaneously.

Lesser the commands, the better it is, WDYT?

Reviewer 2: `kubectl edit` is not something that can be automated easily. I prefer to give the `kubectl annotate` and `kubectl patch` commands as examples, in addition to the description you have now.
Fewer commands is nice, but also expect that automated tools deploy apps and configure extra things like disabling reclaim space for its PVC(s).

Reviewer 3: Please provide the command to annotate the CR as mentioned above
